deployments/arbitrum/usdc/migrations/1679020486_configurate_and_ens.ts

```diff
@@ -0,0 +1,355 @@
+import { BigNumber, Contract, utils } from 'ethers';
+import { expect } from 'chai';
+import { DeploymentManager } from '../../../../plugins/deployment_manager/DeploymentManager';
+import { migration } from '../../../../plugins/deployment_manager/Migration';
+import { calldata, exp, getConfigurationStruct, proposal } from '../../../../src/deploy';
+
+const ENSName = 'compound-community-licenses.eth';
+const ENSResolverAddress = '0x4976fb03C32e5B8cfe2b6cCB31c09Ba78EBaBa41';
+const ENSSubdomainLabel = 'v3-additional-grants';
+const ENSSubdomain = `${ENSSubdomainLabel}.${ENSName}`;
+const ENSTextRecordKey = 'v3-official-markets';
+
+// https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-bridge-eth/contracts/libraries/AddressAliasHelper.sol
+function applyL1ToL2Alias(l1Address: string) {
+  const offset = BigNumber.from('0x1111000000000000000000000000000000001111');
+  return BigNumber.from(l1Address).add(offset).toHexString();
+}
+
+const gatewayInterface = new utils.Interface(
+  [
+    'function counterpartGateway() view external returns (address)',
+    'function finalizeInboundTransfer(address _token, address _from, address _to, uint256 _amount, bytes calldata _data) external payable'
+  ]
+);
+
+async function estimateL2Transaction(
+  { from, to, data }: { to: string, from: string, data: string },
+  l2DeploymentManager: DeploymentManager
+) {
+  // guess what the l1 gas price will be when the proposal is executed
+  const l1GasPrice = (utils.parseUnits('33', 'gwei')).toNumber();
+  // XXX add buffer?
+  const l2GasPrice = (utils.parseUnits('0.1', 'gwei')).toNumber();
+
+  const l2GasEstimateHex = await l2DeploymentManager.hre.network.provider.send(
+    'eth_estimateGas',
+    [{ from, to, data }]
+  );
+  const l2GasEstimate = BigNumber.from(l2GasEstimateHex);
+
+  // Add overhead to cover retryable ticket creation etc
+  const gasBuffer = 0; // XXX
+  const l2GasLimit = BigNumber.from(gasBuffer).add(l2GasEstimate.mul(3).div(2));
+
+  const bytesLength = utils.hexDataLength(data);
+  // https://etherscan.io/address/0x5aed5f8a1e3607476f1f81c3d8fe126deb0afe94#code
+  // calculateRetryableSubmissionFee
+  const submissionCost = (1400 + 6 * bytesLength) * l1GasPrice;
+  const submissionCostWithMargin = utils.parseUnits('10', 'gwei').add(submissionCost);
+
+  const deposit = submissionCostWithMargin.add(l2GasLimit.mul(l2GasPrice));
+
+  // XXX add l2CallValue?
```

Comment:
We expect this to always be 0 right? Only reason it would be populated is if we want to send ETH over the bridge?